nixos/modules/config/pulseaudio.nix

```diff
@@ -118,6 +140,31 @@ in {
           '';
         };
       };
+
+      zeroconf = {
+        discovery.enable =
+          mkEnableOption "discovery of pulseaudio sinks in the local network";
+        publish.enable =
+          mkEnableOption "publishing the pulseaudio sink in the local network";
+      };
+
+      # TODO: enable by default?
+      tcp = {
+        enable = mkEnableOption "tcp streaming support";
+
+        anonymousClients = {
+          allowAll = mkEnableOption "all anonymous clients to stream to the server";
+          allowedIpRanges = mkOption {
+            type = types.listOf types.string;
```

Comment:
types.string is deprecated but still exists. It merges strings together (with no separator), whereas types.str is required be unique (fail if defined multiple times). I guess it doesn't matter in a list, but deprecated is deprecated :-)